awx_collection/plugins/modules/tower_credential.py

```diff
@@ -385,6 +401,28 @@ def main():
     if organization:
         lookup_data['organization'] = org_id
 
+    # Attempt to look up credential to copy based on the provided name
+    if copy_from:
+        # Check if credential exists, as API will allow you to create an identical item with the same name in same org, but GUI will not.
+        credential = module.get_one('credentials', name_or_id=name, **{'data': lookup_data})
+        if credential is not None:
+            module.fail_json(msg="A credential with the name {0} already exists.".format(name))
+        else:
+            # Lookup existing credential.
+            copy_lookup_data = {
+                'credential_type': cred_type_id,
+            }
+            copy_from_lookup = module.get_one('credentials', name_or_id=copy_from, **{'data': copy_lookup_data})
+            if copy_from_lookup is None:
+                module.fail_json(msg="A credential with the name {0} was not able to be found.".format(copy_from))
+            else:
```

Comment:
This seems like a redundant pattern. Can we move this logic of checking the target name and checking that the source item exist into the copy function itself to reduce code in modules?

Reply:
I can try and do that, was debating it myself, but some of the items were specific to each one. Will think about this and see about getting the code simplified into the function.

Reply:
Done